UnitsNet.Tests/QuantityTest.cs

```diff
@@ -171,5 +171,19 @@ public void Types_ReturnsKnownQuantityTypes()
             Assert.Superset(knownQuantities.ToHashSet(), types.ToHashSet());
             Assert.Equal(QuantityCount, types.Length);
         }
+
+        [Fact]
+        public void GetUnitForUnitSystemTest()
+        {
+            Assert.Throws<ArgumentNullException>(() => Length.GetUnitFor((UnitSystem)null));
+            Assert.Equal(LengthUnit.Meter, Length.GetUnitFor(UnitSystem.SI));
+        }
+
+        [Fact]
+        public void GetUnitForBaseUnitsTest()
+        {
+            Assert.Throws<ArgumentNullException>(() => Length.GetUnitFor((BaseUnits)null));
```

Comment:
Here too, I don't like the duality of `BaseUnits` and `UnitSystem`. It smells like the two should merge.